### DISCUSSION TITLE
Supported Python Packages in Code Block

### DISCUSSION THREAD

**Discussion by crazywoola** [langgenius/dify#3580] [Help] [answered]
See this commit https://github.com/langgenius/dify/pull/3567/files

**Comment by crazywoola** [comment, marked as answer]
```python
import json
import datetime
import math
import random
import re
import string
import sys
import time
import traceback
import uuid
import os
import base64
import hashlib
import hmac
import binascii
import collections
import functools
import operator
import itertools
```

**Comment by ricky-algorime** [comment]
how can i add more, using docker compese file

**Reply by crazywoola** [reply]
Cann't do this at this moment.

**Reply by yzning** [reply]
> how can i add more, using docker compese file

Is it possible now？ like adding other packages into sandbox container and importing them in the code block...

> Cann't do this at this moment.
Is it possible now？ like adding other packages into sandbox container and importing them in the code block...

**Reply by johnpccd** [reply]
Hey @crazywoola , is this possible with sandbox's dependecies/python-requirements.txt? I tried to add "requests" into that file, rebuilt the container and restarted, but it still doesn't work.

**Comment by ddnoodle** [comment]
> See this commit https://github.com/langgenius/dify/pull/3567/files

I met the same issue, need to add additional package in code node. how to achieve this?

**Reply by crazywoola** [reply]
Please upgrade the sandbox to 0.2.4 and then you are able to add dependencies in `dependecies/python-requirements.txt`
